modules/account-lib/src/coin/hbar/utils.ts

```diff
@@ -182,3 +183,35 @@ export function removePrefix(prefix: string, key: string): string {
   }
   return key;
 }
+
+/**
+ * Validate if a hedera encoded transaction was signed by a certain
+ * signer and if the signature is the same
+ *
+ * @param {string} encodedTx a hedera encoded transaction
+ * @param {KeyPair} key the signer's keypair
+ * @param {string} signature the signature to be compared
+ * @returns {boolean} the result of the validation
+ */
+export function validate(encodedTx: string, key: KeyPair, signature: string): boolean {
+  let wasSigned = false;
+  const signer = key.getKeys().pub.toString(true);
+  const transaction = SDKTransaction.fromBytes(toUint8Array(encodedTx));
+  const transactionData = transaction._toProto();
+
+  if (transactionData.getSigmap()) {
+    transactionData
```

Comment:
should we check if there's 0 sigs? would that break for this?

Reply:
Hmm, that's a good one, i'll check about this